src/ngraph/op/reverse.cpp

```diff
@@ -69,3 +69,84 @@ void op::Reverse::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVect
 
     adjoints.add_delta(x, make_shared<op::Reverse>(delta, m_reversed_axes));
 }
+
+const string op::v1::Reverse::type_name{"Reverse"};
+
+op::v1::Reverse::Reverse(const Output<Node>& data,
+                         const Output<Node>& reversed_axes,
+                         const std::string& mode)
+    : Op({data, reversed_axes})
+    , m_mode{mode}
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v1::Reverse::validate_and_infer_types()
```

Comment:
If input 1 is fed by a `Constant` (and input 0 rank is static), we should still check to make sure that each element of the axis set is < rank. You can use `Node::is_constant()`, `dynamic_pointer_cast`, and `Constant::get_axis_set_val` for this.

Reply:
(I guess you'll need a separate codepath in case the "mask" behavior is selected, but I think we should still check it.)

Reply:
I've done the validation for the `Constant` second input case and covered it with some tests. I've also added a separate path for the `mask` mode but it turned out it was validated by another check that already exists (checks if the number of values in the mask matches the input rank). It does not operate on the actual values of the constant but rather on shapes. Is this ok?

Reply:
Looks great! For `mask` I think that makes sense---any mask of the correct shape is valid by definition, so no need to check the values in the mask.